.gitlab-ci.yml

```diff
@@ -1,86 +1,25 @@
-# Copyright 2022 ETH Zurich and University of Bologna.
+# Copyright 2024 ETH Zurich and University of Bologna.
 # Licensed under the Apache License, Version 2.0, see LICENSE for details.
 # SPDX-License-Identifier: Apache-2.0
 
 # Author: Sergio Mazzola <smazzola@iis.ee.ethz.ch>
 
-variables:
-  # Enable colors in CI terminal
-  TERM: ansi
-  FORCE_COLOR: 1
-  # Toolchain
-  QUESTA_SEPP: questa-2022.3
-  VSIM: $QUESTA_SEPP vsim
-  RISCV_GCC_BINROOT: /usr/pack/riscv-1.0-kgf/pulp-gcc-2.5.0/bin
-  CC: /usr/pack/gcc-11.2.0-af/linux-x64/bin/gcc
-  CXX: /usr/pack/gcc-11.2.0-af/linux-x64/bin/g++
-  PYTHON: /usr/local/anaconda3-2022.05/bin/python3
-  CMAKE: cmake-3.28.3
-
-before_script:
-  # Set up environment
-  - export PATH=$RISCV_GCC_BINROOT:$PATH
-  - if [ -d .venv ]; then source .venv/bin/activate; fi
-  # Check environment variables
-  - env
+# We initialize the nonfree repo, then spawn a sub-pipeline from it
 
 stages:
-  - setup
-  - build
-  - test
-
-.base:
-  artifacts:
-    when: always
-    expire_in: 1 week
-
-setup-env:
-  extends: .base
-  stage: setup
-  script:
-    # Install python virtual environment
-    - $PYTHON -m venv .venv
-    - source .venv/bin/activate
-    - python3 -m pip install --upgrade pip
-    - python3 -m pip install -r requirements.txt
-  artifacts:
-    paths: [ ".venv" ]
-
-vsim-build:
-  extends: .base
-  stage: build
-  needs: [ setup-env ]
-  script:
-    # Checkout dependencies
-    - bender checkout
-    # Build hardware
-    - make chs-hw-init
-    - make snitch-hw-init
-    - make chs-sim-all
-    - make chim-sim
-    # Compile software
-    - make chim-sw
-    # Compile SoC in vsim
-    - cd target/sim/vsim
-    - $VSIM -c -do 'quit -code [source compile.tcl]'
-  dependencies:
-    - setup-env
-  artifacts:
-    paths: [ ".venv", "hw", "sw", "target/sim" ]
+  - nonfree
 
-vsim-test:
-  extends: .base
-  stage: test
-  needs: [ vsim-build ]
-  parallel:
-    matrix:
-      - TESTNAME: [testCluster, testClusterOffload]
```

Comment:
Done! I think it's better to still decide manually which tests to execute. I also made it in a generic way in case the same approach is required for different usage in the future.
Now there's a `.gitlab` directory which contains the internal GitLab CI manifest (launching the nonfree subpipe), and another yml including only a list of variables which are used by the CI in the nonfree repo.